### PR TITLE
Updating etcdless docs

### LINKED ISSUE #654: Create KDD + Kubeadm-1.6 compliant YAMLS (add RBAC)
http://docs.projectcalico.org/v2.1/getting-started/kubernetes/installation/hosted/k8s-backend/calico.yaml

is referenced from here: http://docs.projectcalico.org/v2.1/getting-started/kubernetes/installation/hosted/k8s-backend/calico.yaml 

But this manifest won't work with k8s 1.6+ because it lacks the RBAC stuff.

### LINKED ISSUE #660: calico/node reference missing KDD connection info
That table should include these env vars as well: http://docs.projectcalico.org/v2.1/reference/calicoctl/setup/kubernetes#complete-list-of-kubernetes-api-connection-configuration

### LINKED ISSUE #661: Rename "etcdless" to "KDD"
We need to rename the "etcdless" install to be "Kubernetes Datastore Driver" as well as update the page to match the features of the latest version:

- [x] Update self-hosted manifest to enable BGP
- [x] Update kubeadm + flannel section to remove flannel
- [x] Update limitations section of "etcdless" doc
- [x] Rename from "etcdless" to "Kubernetes datastore driver"

### PR DESCRIPTION
**Pull request by heschlie**
Fixes #661 
Fixes #660 
Fixes #654 

- Renamed page to "Kubernetes Datastore Driver"
- Updating to be more inline with new BGP features in KDD
  - Removed Flannel references in kbeadm section
  - Removed routing limitation
- Updated calico.yaml manifest to use BGP routing

### CODE REVIEW

**Comment by djosborne** [comment]
@heschlie I think we should go beyond changing the title and change the filepath as well. From:
|----|-----|
| From: | `master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md` |
| To: | `master/getting-started/kubernetes/installation/hosted/kubernetes-datastore-driver` |

We can use `redirect_from` for 1 release to move people to the new page as well.

Thoughts?

File: master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md [hunk 1]
```diff
@@ -77,7 +75,7 @@ export KUBE_MASTER_OS_DISTRIBUTION=debian
 curl -sS https://get.k8s.io | bash
 ```
 
-##### 2) Install Calico using the etcdless manifest
+##### 2) Install Calico using the Kubenetes datastore manifest
```
Comment (djosborne) [inline master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md]: Datastore driver? Or just Datastore?

**Comment by heschlie** [comment]
Makes sense and would be less confusing for people looking for the URL.

File: master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md (hunk 1 above)
Comment (heschlie) [inline master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md]: Should be Datastore driver

Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md]: I'm not sure we want to be calling it the Kubernetes datastore driver.  I think "Kubernetes datastore manifest" is sufficient.

The "Kubernetes datastore driver" is a specific piece of code.

File: master/getting-started/kubernetes/installation/hosted/index.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ the recommended hosted approach for deploying Calico in production.
 This manifest installs Calico as well as a single node etcd cluster.  This is the recommended hosted approach
 for getting started quickly with Calico in conjunction with tools like kubeadm.
 
-#### [Etcdless Hosted Install](k8s-backend/)
+#### [Kubernetes Datastore Driver](k8s-backend/)
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/index.md]: I think this should read `Kubernetes Datastore Hosted Install`

(and other references to "Kubernetes datastore driver" should probably just be "Kubernetes datastore")

File: master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md [hunk 3]
```diff
@@ -1,12 +1,11 @@
 ---
-title: Etcdless Hosted Install
+title: Kubernetes Datastore Driver
 ---
 
 This document describes installing Calico on Kubernetes in a mode that does not require access to an etcd cluster.  
 This mode uses the Kubernetes API as the datastore.  Note that this feature
 currently comes with a number of limitations, namely:
 
-- it supports policy enforcement only and does not yet support Calico BGP networking.
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/k8s-backend/index.md]: This should now read "It supports BGP full-mesh networking, but does not yet support BGP peer configuration"

**Comment by caseydavenport** [comment]
We probably want to tweak the [Integration guide requirements section](http://docs.projectcalico.org/master/getting-started/kubernetes/installation/integration#requirements) as well.

**Comment by caseydavenport** [comment]
@heschlie could you update this PR with the status of re-testing the two recommended install methods?

i.e.

- [ ] kubeadm
- [ ] kube-up on GCE

**Comment by heschlie** [comment]
@caseydavenport @djosborne I updated the env var table in the calico/node reference, but I was wondering if we should denote that they can be prefixed with `CALICO_` somehow?  On the calicoctl/k8s page we have it as a note at the bottom of the table, but that would be confusing on this page, thoughts?

**Comment by frnkdny** [comment]
@heschlie regarding the install method: kube-up on GCE
- `KUBE_MASTER_OS_DISTRIBUTION=debian` support has been removed for debian on master in GCE (cannot operate on cluster using master os distro: debian)
- After looking further and exploring using just the default value I noticed that the Cluster Configuration directory has a Deprecation Notice: This directory has entered maintenance mode https://github.com/kubernetes/kubernetes/blob/master/cluster/README.md
- The new recommendation is to kube-deploy https://github.com/kubernetes/kube-deploy

After a quick chat with @caseydavenport the suggestion is to update these docs for kube-deploy rather than kube-up on GCE.

**Comment by caseydavenport** [comment]
> After a quick chat with @caseydavenport the suggestion is to update these docs for kube-deploy rather than kube-up on GCE.

Looking at kube-deploy, it does't seem to yet provide equivalent functionality to what we used before.

We may want to continue with kube-up (it doesn't seem to be going away just yet) until we've found an equivalent solution on GCE.

**Comment by heschlie** [comment]
@caseydavenport Do you feel we need to elaborate on the `kubeadm` section more if we're removing the GCE section?

File: master/getting-started/kubernetes/installation/hosted/index.md (hunk 2 above)
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/index.md]: ping

File: master/getting-started/kubernetes/installation/hosted/k8s-backend/calico.yaml [hunk 4]
```diff
@@ -70,9 +70,6 @@ spec:
             # Enable felix debug logging.
             - name: FELIX_LOGSEVERITYSCREEN
               value: "debug"
-            # Don't enable BGP.
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/k8s-backend/calico.yaml]: Let's also enable IPIP by default in this manifest.

**Comment by caseydavenport** [comment]
@heschlie I don't think so.

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/1.6/calico.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+              value: "true"
+            # Set Felix endpoint to host default action to ACCEPT.
+            - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
+              value: "ACCEPT"
+            # Disable IPV6 on Kubernetes.
+            - name: FELIX_IPV6SUPPORT
+              value: "false"
+            # Wait for the datastore.
+            - name: WAIT_FOR_DATASTORE
+              value: "false"
```
Comment (djosborne) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/1.6/calico.yaml]: Now that https://github.com/projectcalico/calicoctl/issues/1596 is resolved we should be able to set this back to `true`.

**Pull request merged**